playbooks/libvirt/openshift-cluster/tasks/configure_libvirt_network.yml

```diff
@@ -1,27 +1,10 @@
 ---
-- name: Test if libvirt network for openshift already exists
-  command: "virsh -c {{ libvirt_uri }} net-info {{ libvirt_network }}"
-  register: net_info_result
-  changed_when: False
-  failed_when: "net_info_result.rc != 0 and 'no network with matching name' not in net_info_result.stderr"
-
-- name: Create a temp directory for the template xml file
-  command: "mktemp -d /tmp/openshift-ansible-XXXXXXX"
-  register: mktemp
-  when: net_info_result.rc == 1
-
-- name: Create network xml file
-  template:
-    src: templates/network.xml
-    dest: "{{ mktemp.stdout }}/network.xml"
-  when: net_info_result.rc == 1
-
-- name: Create libvirt network for openshift
-  command: "virsh -c {{ libvirt_uri }} net-create {{ mktemp.stdout }}/network.xml"
-  when: net_info_result.rc == 1
-
-- name: Remove the temp directory
-  file:
-    path: "{{ mktemp.stdout }}"
-    state: absent
-  when: net_info_result.rc == 1
+- name: Create the libvirt network for OpenShift
+  virt_net:
+    name: '{{ libvirt_network }}'
+    state: '{{ item }}'
+    xml: "{{ lookup('template', 'network.xml') }}"
+    uri: '{{ libvirt_uri }}'
```

Comment:
Indeed, makes sense.
Note that I had to quote the `yes` in `autostart: 'yes'` because without the quotes (i.e.: `autostart: yes`), I got the following error:

```
TASK [Create the libvirt storage pool for OpenShift] ***************************
failed: [localhost] (item=present) => {"failed": true, "item": "present", "msg": "value of autostart must be one of: yes,no, got: True"}
failed: [localhost] (item=active) => {"failed": true, "item": "active", "msg": "value of autostart must be one of: yes,no, got: True"}
```

which seems to have already been fixed upstream by https://github.com/ansible/ansible-modules-extras/pull/2593/files#diff-00d13769f5d06664a824c39ec04d393c . My understanding of the error message is that the yaml parser interprets the `yes` as a boolean and the ansible module expected the string `yes`.